pkg/broker/handler/processors/deliver/processor.go

```diff
@@ -239,6 +239,9 @@ func (p *Processor) deliver(ctx context.Context, target *config.Target, broker *
 	if err != nil {
 		return err
 	}
+	if replyResp.StatusCode/100 != 2 {
```

Comment:
We should probably add some metrics around these replies too...

Reply:
More metrics is already better. Can we change this to sth like `replyResp.StatusCode >= 200 && replyResp.StatusCode < 300`, I found this easier to process and it is also faster for the CPU.

Reply:
Added a TODO, making this metrics is a much larger change.